File: backend/app/schemas/analytics.py
from typing import List
from pydantic import BaseModel, Field

class RecoveryTotalsResponse(BaseModel):
    total_cases: int = Field(default=0, description="Total number of closed cases")
    total_gross_recovered: float = Field(default=0.0, description="Total gross revenue recovered")
    total_intervention_costs: float = Field(default=0.0, description="Total cost of all interventions")
    total_discounts: float = Field(default=0.0, description="Total amount of discounts offered")
    total_net_recovered: float = Field(default=0.0, description="Total net revenue recovered")

class InterventionStat(BaseModel):
    intervention_type: str = Field(description="The type of intervention")
    count: int = Field(default=0, description="Number of times this intervention was executed")
    total_cost: float = Field(default=0.0, description="Total cost for this intervention type")

class InterventionStatsResponse(BaseModel):
    stats: List[InterventionStat] = Field(default_factory=list, description="Statistics grouped by intervention type")

class BaselineMetrics(BaseModel):
    gross_recovered: float = Field(default=0.0)
    net_recovered: float = Field(default=0.0)
    recovery_rate: float = Field(default=0.0)
    escalation_rate: float = Field(default=0.0)

class BaselineComparisonResponse(BaseModel):
    simulation_run_id: str
    seed: int
    baseline: BaselineMetrics = Field(description="Metrics for naive baseline strategy")
    revive: BaselineMetrics = Field(description="Metrics for REVIVE agent strategy")
    delta_net: float = Field(description="Difference in net revenue (REVIVE - Baseline)")
    delta_escalation_rate: float = Field(description="Difference in escalation rate (REVIVE - Baseline)")

File: frontend/src/api/analytics.ts
import { apiFetch } from './client';

export interface BaselineMetrics {
  gross_recovered: number;
  net_recovered: number;
  recovery_rate: number;
  escalation_rate: number;
}

export interface BaselineComparisonResponse {
  simulation_run_id: string;
  seed: number;
  baseline: BaselineMetrics;
  revive: BaselineMetrics;
  delta_net: number;
  delta_escalation_rate: number;
}

export interface RecoveryTotalsResponse {
  total_cases: number;
  total_amount_at_risk: number;
  total_gross_recovered: number;
  total_intervention_costs: number;
  total_discounts: number;
  total_net_recovered: number;
  recovery_rate: number;
  average_recovery_time_hours: number | null;
}

export interface InterventionStat {
  intervention_type: string;
  count: number;
  success_count: number;
  total_cost: number;
}

export interface InterventionStatsResponse {
  stats: InterventionStat[];
}

function withQuery(path: string, params: Record<string, string | undefined>): string {
  const search = new URLSearchParams();
  for (const [key, value] of Object.entries(params)) {
    if (value !== undefined) search.append(key, value);
  }
  const qs = search.toString();
  return qs ? `${path}?${qs}` : path;
}

export function fetchBaselineComparison(seed?: number): Promise<BaselineComparisonResponse> {
  const query = seed !== undefined ? `?seed=${seed}` : '';
  return apiFetch(`/analytics/baseline${query}`, undefined, 'Failed to fetch baseline comparison');
}

export function fetchRecoveryTotals(
  startDate?: string,
  endDate?: string,
  caseType?: string,
  origin?: string,
): Promise<RecoveryTotalsResponse> {
  const url = withQuery('/analytics/recovery', {
    start_date: startDate,
    end_date: endDate,
    case_type: caseType,
    origin: origin,
  });
  return apiFetch(url, undefined, 'Failed to fetch recovery totals');
}

export function fetchInterventionStats(
  startDate?: string,
  endDate?: string,
  caseType?: string,
  origin?: string,
): Promise<InterventionStatsResponse> {
  const url = withQuery('/analytics/interventions', {
    start_date: startDate,
    end_date: endDate,
    case_type: caseType,
    origin: origin,
  });
  return apiFetch(url, undefined, 'Failed to fetch intervention stats');
}
